### PR TITLE
Keylight v2 

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/16808

## Description
A rewriting of keylight with additional fetch incidents.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/edit/keylight/Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5008)
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/keylight/Integrations/Lockpath_KeyLight_v2/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5008)
- [ ] [Script YAML] (https://github.com/demisto/content/edit/keylight/Scripts/KeylightCreateRecord/KeylightCreateIssue.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5008)
- [ ] [Script python file] (https://github.com/demisto/content/edit/keylight/Scripts/KeylightCreateRecord/KeylightCreateIssue.py?pr=%2Fdemisto%2Fcontent%2Fpull%2F5008)

### CODE REVIEW

File: Integrations/Lockpath_KeyLight_v2/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## [Unreleased]
```
Comment (yaakovi) [inline Integrations/Lockpath_KeyLight_v2/CHANGELOG.md]: no need to add changelogs for new integrations.

File: Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.yml [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+      type: String
+    - contextPath: Keylight.Record.DisplayName
+      description: The display name of the record.
+      type: String
+    - contextPath: Keylight.Record.Fields
+      description: The fields in the record.
+      type: Unknown
+    - contextPath: Keylight.Record.ID
+      description: The record ID.
+      type: Unknown
```
Comment (yaakovi) [inline Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.yml]:
```suggestion
      type: String
```

File: Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+            demisto.results('ok')
+
+        logged_in = client.login(username, password)
+        if logged_in:
+            commands[demisto.command()](client, demisto.args())
+            client.logout()
+    except Exception as e:
+        LOG.print_log()
+        if logged_in:
+            client.logout()
```
[the remark this answers is not in the document]
Comment (orlichter1) [reply Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py]: does `return_error` exit the code? If `finally` is done after will it still run?

File: Integrations/Lockpath_KeyLight_v2/CHANGELOG.md (hunk 1 above)
Comment (orlichter1) [reply Integrations/Lockpath_KeyLight_v2/CHANGELOG.md]: just remove the the file?

File: Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py]: finally will run even if `return_error` is being called

File: Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+        'kl-update-record': update_record_command,
+        'kl-get-lookup-report-column-fields': get_lookup_report_column_fields_command,
+        'kl-get-record-attachment': get_record_attachment_command,
+        'kl-get-record-attachments': get_record_attachments_command,
+        'kl-delete-record-attachment': delete_record_attachment_command,
+        'fetch-incidents': fetch_incidents
+    }
+
+    LOG(f'Command being called is {demisto.command()}')
+    logged_in = False
```
[the remark this answers is not in the document]
Comment (yaakovi) [reply Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py]: ?

File: Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py [hunk 5]
```diff
@@ -0,0 +1,673 @@
...
+    try:
+        if demisto.command() == 'test-module':
+            client.login(username, password)
+            demisto.results('ok')
+
+        logged_in = client.login(username, password)
+        if logged_in:
+            commands[demisto.command()](client, demisto.args())
+    except Exception as e:
+        LOG.print_log()
```
Comment (yaakovi) [inline Integrations/Lockpath_KeyLight_v2/Lockpath_KeyLight_v2.py]: redundant

**Comment by orlichter1** [comment]
@kirbles19 could you please have a look at the files (2 YMLS and one python with an explanation in it)

**Comment by richardbluestone** [comment]
Docs reviewed and updated where necessary.

**Comment by yaakovi** [comment]
failed on parallel docker, force-pushing this.
will be fixed on another branch

**Pull request merged**